test_programs/compile_success_empty/comptime_module/src/main.nr

```diff
@@ -125,3 +125,79 @@ fn as_module_test() {
     }
 }
 // docs:end:as_module_example
+
+// docs:start:parent_example
+mod module1 {
+    pub mod module2 {}
+}
+
+#[test]
+fn parent_test() {
+    comptime {
+        let my_module2 = quote [module1::module2].as_module().unwrap();
+        assert_eq(my_module2.name(), quote [module2]);
+
+        let my_module1 = my_module2.parent().unwrap();
+        assert_eq(my_module1.name(), quote [module1]);
+
+        // The top-level module in each crate has no name
+        let top_level_module = my_module1.parent().unwrap();
+        assert_eq(top_level_module.name(), quote []);
+    }
+}
+// docs:end:parent_example
+
+// docs:start:child_modules_example
+mod my_module {
+    pub mod child1 {}
+    pub mod child2 {}
+    pub mod child3 {
+        pub mod nested_child {}
+    }
+}
+
+#[test]
+fn child_modules_test() {
+    comptime {
+        let my_module = quote [my_module].as_module().unwrap();
+        let children = my_module.child_modules().map(Module::name);
+
+        // The order children are returned in is left unspecified.
+        assert_eq(children, &[quote [child1], quote [child2], quote [child3]]);
```

Comment:
If it's unspecified, shouldn't this compare by sorting? Or maybe it's unspecified but the current implementation returns them in order so this test is fine?